buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/NamingConventionsTask.groovy

```diff
@@ -81,11 +80,18 @@ public class NamingConventionsTask extends LoggedExec {
         inputs.files(classpath)
         description = "Tests that test classes aren't misnamed or misplaced"
         executable = new File(project.runtimeJavaHome, 'bin/java')
-        if (false == checkForTestsInMain) {
+
+        def dirsToUse = (checkForTestsInMain ?
+            project.sourceSets.main.output.classesDirs :
+            project.sourceSets.test.output.classesDirs)
```

Comment:
I don't think we can check these here, this would still be during configuration? Certainly not the existence of them in the filter below.

Reply:
This works because the FileCollection is a live collection, this the checks are actually only executed lather.
I'm changing this anyhow, as  I discovered that we should not pick up non Java tests anyhow, because those don't adhere to the naming conventions.